contracts/auctions/src/utils.rs

```diff
@@ -0,0 +1,109 @@
+use soroban_sdk::{log, Env};
+
+use crate::{
+    collection,
+    error::ContractError,
+    storage::{
+        save_auction_by_id, save_auction_by_seller, update_auction, Auction, AuctionStatus,
+        HighestBid,
+    },
+    token,
+};
+
+pub(crate) fn check_auction_can_be_finalized(
+    env: &Env,
+    auction: &Auction,
+) -> Result<(), ContractError> {
+    if auction.status != AuctionStatus::Active {
+        log!(
+            env,
+            "Auction: Finalize auction: Cannot finalize an inactive/ended auction."
+        );
+        return Err(ContractError::AuctionNotActive);
+    }
+    if env.ledger().timestamp() < auction.end_time {
+        log!(
+            env,
+            "Auction: Finalize auction: Auction cannot be ended early"
+        );
+        return Err(ContractError::AuctionNotFinished);
+    }
+    Ok(())
+}
+
+pub(crate) fn minimum_price_reached(auction: &Auction) -> bool {
+    !auction
+        .item_info
+        .minimum_price
+        .and_then(|min_price| {
+            auction
+                .highest_bid
+                .map(|highest_bid| min_price > highest_bid)
+        })
+        .unwrap_or(false)
+}
```

Comment:
This is confusing. You are checking if `minimum_price_reached` while really you check if the minimum price has not been reached.

```suggestion
pub(crate) fn minimum_price_reached(auction: &Auction) -> bool {
    auction
        .item_info
        .minimum_price
        .map_or(true, |min_price| {
            auction
                .highest_bid
                .map_or(false, |highest_bid| highest_bid >= min_price)
        })
}
```

- if there's no minimum_price, we assume the minimum price has been reached (since there's no minimum).
- if there's no highest_bid, the minimum price has not been reached.
- the final comparison checks whether the highest_bid is greater than or equal to the minimum_price.